Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp

```diff
@@ -259,14 +259,46 @@ ActsExamples::ProcessCode ActsExamples::SeedingAlgorithm::execute(
       const auto& collection = spacePointsGrouping.grid().at(grid_glob_bin);
       for (const auto& sp : collection) {
         std::size_t index = sp->index();
-        state.spacePointData.setTopHalfStripLength(
-            index, m_cfg.seedFinderConfig.getTopHalfStripLength(sp->sp()));
-        state.spacePointData.setBottomHalfStripLength(
-            index, m_cfg.seedFinderConfig.getBottomHalfStripLength(sp->sp()));
-        state.spacePointData.setTopStripDirection(
-            index, m_cfg.seedFinderConfig.getTopStripDirection(sp->sp()));
-        state.spacePointData.setBottomStripDirection(
-            index, m_cfg.seedFinderConfig.getBottomStripDirection(sp->sp()));
+
+        const float& topHalfStripLength =
+            m_cfg.seedFinderConfig.getTopHalfStripLength(sp->sp());
+        const float& bottomHalfStripLength =
+            m_cfg.seedFinderConfig.getBottomHalfStripLength(sp->sp());
+        const Acts::Vector3& topStripDirection =
```

Comment:
all these function return objects by value, you cannot use `const Acts::Vector3&`

Reply:
I am confused on this point, don't getTopStripDirection and getBottomStripDirection return a Vector3?

Reply:
I'm suggesting to change this to the following:
```cpp
        const float topHalfStripLength =
            m_cfg.seedFinderConfig.getTopHalfStripLength(sp->sp());
        const float bottomHalfStripLength =
            m_cfg.seedFinderConfig.getBottomHalfStripLength(sp->sp());
        const Acts::Vector3 topStripDirection =
            m_cfg.seedFinderConfig.getTopStripDirection(sp->sp());
        const Acts::Vector3 bottomStripDirection =
            m_cfg.seedFinderConfig.getBottomStripDirection(sp->sp());

        state.spacePointData.setTopStripVector(
            index, topHalfStripLength * topStripDirection);
        state.spacePointData.setBottomStripVector(
            index,
            bottomHalfStripLength * bottomStripDirection);
        state.spacePointData.setStripCenterDistance(
            index, m_cfg.seedFinderConfig.getStripCenterDistance(sp->sp()));
        state.spacePointData.setTopStripCenterPosition(
            index, m_cfg.seedFinderConfig.getTopStripCenterPosition(sp->sp()));
```

Reply:
ah of course, sorry